### PR TITLE
added all globals for full jshint compatibility

### LINKED ISSUE #4: Convince JSHint to use this module
Would be nice to have parity between them.

_Now included in ESLint. Still need to convince JSHint though._

### PR DESCRIPTION
**Pull request by valorkin**
resolve #4 
added all globals for full jshint compatibility

### CODE REVIEW

File: globals.json [hunk 1]
```diff
@@ -557,5 +562,158 @@
...
+		"prompt": false
+	},
+	"dojo": {
+		"dojo": false,
+		"dijit": false,
+		"dojox": false,
+		"define": false,
+		"require": false
+	},
+	"typed": {
```
Comment (nzakas) [inline globals.json]: Already part of `builtin`

File: globals.json [hunk 2]
```diff
@@ -557,5 +562,158 @@
...
+		"__filename": false,
+		"__dirname": false,
+		"global": false,
+		"module": false,
+		"require": false,
+		"Buffer": true,
+		"exports": true,
+		"process": true
+	},
+	"devel": {
```
Comment (nzakas) [inline globals.json]: This was just merged into `browser` in the last release.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "globals",
-  "version": "5.0.0",
+  "version": "5.1.0",
```
Comment (nzakas) [inline package.json]: Please don't update the package version. This is done as part of the release process.

File: globals.json [hunk 4]
```diff
@@ -557,5 +562,158 @@
...
+		"Draggable": false,
+		"Draggables": false,
+		"Droppables": false,
+		"Effect": false,
+		"Sortable": false,
+		"SortableObserver": false,
+		"Sound": false,
+		"Scriptaculous": false
+	},
+	"browserify": {
```
Comment (nzakas) [inline globals.json]: Aren't these already covered by the `node` environment?

File: globals.json [hunk 5]
```diff
@@ -416,7 +416,8 @@
 	"worker": {
 		"importScripts": true,
 		"postMessage": true,
-		"self": true
+		"self": true,
+		"FileReaderSync": true
 	},
 	"node": {
```
Comment (michaelficarra) [inline globals.json]: None of these additions to `node` are actually globals in node (at least as of `0.11.14`).

**Comment by michaelficarra** [comment]
I don't think "resolve #17" is correct, since this *is* #17.

File: globals.json (hunk 4 above)
Comment (valorkin) [inline globals.json]: @nzakas they may intersect, but it is done for jshint compliance
https://github.com/jshint/jshint/blob/master/src/vars.js
related PR: https://github.com/eslint/eslint/pull/1690

**Comment by valorkin** [comment]
@michaelficarra agreed, it is about to resolve #4 
and forget about projects which support 0.9+

File: globals.json (hunk 2 above)
Comment (valorkin) [inline globals.json]: @nzakas yeap I saw, but the reason I did is to pair jshint

**Comment by es128** [comment]
:-1: this wouldn't actually resolve #4 without a commitment from jshint maintainers to use this list. 

The goal of #4 isn't quite about achieving parity with jshint's current list. It's about creating the most sensible list possible, then convincing jshint that it's in their best interest to use it despite the differences.

**Comment by michaelficarra** [comment]
I fully agree with @es128. :-1:

**Comment by nzakas** [comment]
:-1: Totally agree with @es128.

**Comment by es128** [comment]
Alright, I'll do the honors.

**Comment by es128** [comment]
By the way, @valorkin your effort here is appreciated, albeit a bit misguided. Perhaps you can shift your attention to the jshint repo to see how they feel about the possibility of moving in this direction. 

Perhaps you can create a bridge solution for a PR on that side that uses this repo then adjusts it for parity with what they already have. See https://github.com/eslint/eslint/pull/1586 for a sort-of example.

**Comment by sindresorhus** [comment]
> The goal of #4 isn't quite about achieving parity with jshint's current list. It's about creating the most sensible list possible, then convincing jshint that it's in their best interest to use it despite the differences.

:+1: Exactly.

> Perhaps you can create a bridge solution for a PR on that side that uses this repo then adjusts it for parity with what they already have. See eslint/eslint#1586 for a sort-of example.

:+1:

**Comment by valorkin** [comment]
@es128 to make PR to jshint I need this repo to be paired to what they already have,
or didn't?

**Comment by es128** [comment]
@valorkin I'm suggesting that the jshint PR uses this module, then mutates the object to match what they already have as a first step that causes no functional changes. Then they can gradually phase out the differences that don't make sense. And if they want to keep sections like `mootools` and `dojo` for backward compatibility, they can make that choice too. But the core list would be maintained over here.

**Comment by valorkin** [comment]
@es128 most probably it will not happen
eslint have 17k loads this week
jshint have 253k loads this week
I hope it will change later, but now...

"dictate the rules" when you are smaller has no sense usually

**Comment by es128** [comment]
It's up to you. I was just making a suggestion. Personally, I care very little whether jshint ever comes around.

> when you are smaller

How about when you're right? This is open-source, everyone starts smaller.

**Comment by valorkin** [comment]
@es128 
agreed, what care actually is https://github.com/eslint/eslint/pull/1690#issuecomment-70817320
`phantom`,` jquery`, `prototypejs`, `shelljs`

I am using this stuff and I would love to have it globals natively,
so can we add at least this ones? (just say yes and PR will be there at the same time :) )

**Comment by sindresorhus** [comment]
Yeah, never mind for now. We can look into convincing JSHint when we've perfected this list.

**Comment by es128** [comment]
@valorkin I think I'd be ok with the extra sections if they'd actually get used. Go ahead and open the PR so the maintainers can discuss it in that context.

**Pull request closed without merging**